src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultRazorDynamicFileInfoProvider.cs

```diff
@@ -144,14 +144,22 @@ public void PromoteBackgroundDocument(Uri documentUri, IRazorDocumentPropertiesS
             throw new ArgumentNullException(nameof(propertiesService));
         }
 
-        // TODO: This needs to use the project key somehow, rather than assuming all generated content is the same
+        // TODO: Get the project info out of the documentUri so that the DocumentFilePathProvider can reconstruct the ProjectKey
         var filePath = GetProjectSystemFilePath(documentUri);
         foreach (var associatedKvp in GetAllKeysForPath(filePath))
         {
             var associatedKey = associatedKvp.Key;
             var associatedEntry = associatedKvp.Value;
 
-            var filename = _documentFilePathProvider.GetRazorCSharpFilePath(associatedKey.FilePath);
+            var projectId = associatedKey.ProjectId;
+            var projectKey = TryFindProjectKeyForProjectId(projectId);
+            if (projectKey is not ProjectKey key)
+            {
+                Debug.Fail("Could not find project key for project id. This should never happen.");
```

Comment:
What is the end user experience in this scenario? Should we be throwing an exception here instead and/or logging? (same question for the VS Mac class)